### ISSUE TITLE
Continuous data and discontinuous results

### ISSUE THREAD

**Issue by zsxgb** [CcGAN2020/CcGAN#1]
I think you proposed Continuous Conditional GAN, this technique is very interesting.

But I would like to ask you, in the simulation you proposed, shouldn't continuous labels get continuous data?
Why do continuous data get discontinuous results?

**Comment by CcGAN2020** [comment]
Hi,
Thanks for your interest in our work.

For your question, which discontinuity do you refer to? In the CIRCULAR 2-D GAUSSIANS, each label corresponds to a conditional Gaussian so data from this conditional Gaussian for sure are continuous.

**Comment by zsxgb** [comment]
![8500](https://user-images.githubusercontent.com/14177065/101767262-f0cdaa00-3b1e-11eb-8dc9-2db8d2c940ba.png)
In this picture, the blue points are the training data, and these blue points are the NetG label (discrete), corresponding to a specific angle (netG training data (discrete)),
The green dots are netG generated data (discrete), corresponding to a specific angle (netG generated data (continuous)).
Is this reasonable?

**Comment by CcGAN2020** [comment]
Yes, it reasonable. This figure is just a visualization of how does CcGAN perform on 10 test angels (which may not be used for training). In our experiment, we test our GAN on 360 angles and it is impossible to visualize all 360 angles since it will be too crowded. The training data are from 120 Gaussians evenly distributed on the unit circle, not just 10 Gaussians. Note that in practice, it is impossible to generate truly "continuous" data which is one of the motivations of CcGAN. Please refer to 4.1 CIRCULAR 2-D GAUSSIANS of our paper for detailed training/test setups.

**Comment by CcGAN2020** [comment]
BTW, the data from each conditional Gaussian are definitely continuous due to the definition of a Gaussian random variable


for each angle (i.e., the mean for the corresponding conditional Gaussian), its data are definitely continuous and should be around this angle ()

**Comment by CcGAN2020** [comment]
BTW, I think the figure you show is the one generated during training which is used to show the performance of CcGAN on 10 test angles.  In the evaluation phase, we can generate samples from any Gaussians (any location on the circle and any number of Gaussians), these test angles may not be used in the training.
